frontend/dockerfile/instructions/commands_runmount.go

```diff
@@ -151,13 +156,29 @@ func parseMount(value string) (*Mount, error) {
 		}
 
 		value := parts[1]
+		// check for potential variable
+		if expander != nil {
```

Comment:
You can just do the `from` check in here without requiring per-key checks. I think we should also not try at expansion at all for fields we ignore the error. It's kind of a weird logic where we call function knowing it will fail. Unless we do it in some other way we should not attempt on the first try. The fields except `from` are not allowed to be used after first pass with this logic. With ignoring the errors this might go unnoticed.

DCO+squash

Reply:
This is actually what I did initially, but I thought you'd prefer to handle it on the key level. Anyway, I'll push another iteration in a few